api-extractor/src/IDocItem.ts

```diff
@@ -141,7 +141,8 @@ export interface IDocMethod extends IDocBase {
   /**
    * a mapping of parameter name to IDocParam
    */
-  parameters: Map<string, IDocParam>;
+
+  parameters: { [name: string]: IDocParam};
```

Comment:
Using a JSO map instead of Map requires every caller to call Object.hasOwnProperty() before accessing it.
Could we add some unit test cases to check these cases, something like this?

```
class __proto__ {
    public propertyIsEnumerable: string;
}

interface hasOwnProperty {
    __lookupSetter__: __proto__;
}

class A extends __proto__ implements hasOwnProperty {
    __lookupSetter__: __proto__;
    public __proto__(__proto__: string): __proto__ {
        return undefined;   
    }
}
```